Source/Modules/r.cxx

```diff
@@ -2141,7 +2145,18 @@ int R::functionWrapper(Node *n) {
   if (destructor)
     Printv(f->code, "R_ClearExternalPtr(self);\n", NIL);
 
-  Printv(f->code, "return r_ans;\n}\n", NIL);
+  Printv(f->code, "return r_ans;\n", NIL);
+  
+  /* Error handling code */
+  Printv(f->code, "fail: SWIGUNUSED;\n", NIL);
+  if (need_cleanup) {
+    Printv(f->code, cleanup, NIL);
+  }
+  Printv(f->code, "  Rf_error(\"%s %s\", SWIG_ErrorType(SWIG_lasterror_code), SWIG_lasterror_msg);\n", NIL);
```

Comment:
I looked for the usage of the functions `warning()`and `stop()`in R that use internally `Rf_warning()` and `Rf_error()` respectively. I found this detailed article: "[A Warning About warning()](https://www.r-bloggers.com/a-warning-about-warning/)". This may help us to decide what to choose between `Rf_warning()`and `Rf_error()`.